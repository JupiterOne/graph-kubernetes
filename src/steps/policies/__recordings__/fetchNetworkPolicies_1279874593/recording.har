{
  "log": {
    "_recordingName": "fetchNetworkPolicies",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "6.0.5"
    },
    "entries": [
      {
        "_id": "e262abbd5b6acc4c1139793afa880ed2",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "name": "accept",
              "value": "application/json"
            },
            {
              "name": "host",
              "value": "192.168.49.2:8443"
            }
          ],
          "headersSize": 115,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "limit",
              "value": "250"
            }
          ],
          "url": "https://192.168.49.2:8443/api/v1/namespaces?limit=250"
        },
        "response": {
          "bodySize": 2052,
          "content": {
            "mimeType": "application/json",
            "size": 2052,
            "text": "{\"kind\":\"NamespaceList\",\"apiVersion\":\"v1\",\"metadata\":{\"resourceVersion\":\"51782\"},\"items\":[{\"metadata\":{\"name\":\"default\",\"uid\":\"868d4703-b338-4051-98e4-b65b4f0a0006\",\"resourceVersion\":\"39\",\"creationTimestamp\":\"2024-01-11T15:50:15Z\",\"labels\":{\"kubernetes.io/metadata.name\":\"default\"},\"managedFields\":[{\"manager\":\"kube-apiserver\",\"operation\":\"Update\",\"apiVersion\":\"v1\",\"time\":\"2024-01-11T15:50:15Z\",\"fieldsType\":\"FieldsV1\",\"fieldsV1\":{\"f:metadata\":{\"f:labels\":{\".\":{},\"f:kubernetes.io/metadata.name\":{}}}}}]},\"spec\":{\"finalizers\":[\"kubernetes\"]},\"status\":{\"phase\":\"Active\"}},{\"metadata\":{\"name\":\"kube-node-lease\",\"uid\":\"52615a92-75c3-4c4b-8b72-51be82a4bc21\",\"resourceVersion\":\"35\",\"creationTimestamp\":\"2024-01-11T15:50:15Z\",\"labels\":{\"kubernetes.io/metadata.name\":\"kube-node-lease\"},\"managedFields\":[{\"manager\":\"kube-apiserver\",\"operation\":\"Update\",\"apiVersion\":\"v1\",\"time\":\"2024-01-11T15:50:15Z\",\"fieldsType\":\"FieldsV1\",\"fieldsV1\":{\"f:metadata\":{\"f:labels\":{\".\":{},\"f:kubernetes.io/metadata.name\":{}}}}}]},\"spec\":{\"finalizers\":[\"kubernetes\"]},\"status\":{\"phase\":\"Active\"}},{\"metadata\":{\"name\":\"kube-public\",\"uid\":\"7a2ee4f7-12fd-494b-93f0-c639b1e63a72\",\"resourceVersion\":\"20\",\"creationTimestamp\":\"2024-01-11T15:50:15Z\",\"labels\":{\"kubernetes.io/metadata.name\":\"kube-public\"},\"managedFields\":[{\"manager\":\"kube-apiserver\",\"operation\":\"Update\",\"apiVersion\":\"v1\",\"time\":\"2024-01-11T15:50:15Z\",\"fieldsType\":\"FieldsV1\",\"fieldsV1\":{\"f:metadata\":{\"f:labels\":{\".\":{},\"f:kubernetes.io/metadata.name\":{}}}}}]},\"spec\":{\"finalizers\":[\"kubernetes\"]},\"status\":{\"phase\":\"Active\"}},{\"metadata\":{\"name\":\"kube-system\",\"uid\":\"ce86116b-863b-4e74-979e-3c031a9aeee1\",\"resourceVersion\":\"12\",\"creationTimestamp\":\"2024-01-11T15:50:15Z\",\"labels\":{\"kubernetes.io/metadata.name\":\"kube-system\"},\"managedFields\":[{\"manager\":\"kube-apiserver\",\"operation\":\"Update\",\"apiVersion\":\"v1\",\"time\":\"2024-01-11T15:50:15Z\",\"fieldsType\":\"FieldsV1\",\"fieldsV1\":{\"f:metadata\":{\"f:labels\":{\".\":{},\"f:kubernetes.io/metadata.name\":{}}}}}]},\"spec\":{\"finalizers\":[\"kubernetes\"]},\"status\":{\"phase\":\"Active\"}}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "audit-id",
              "value": "4270dfd0-7d0b-4d6a-9474-b9709067f9d7"
            },
            {
              "name": "cache-control",
              "value": "no-cache, private"
            },
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "x-kubernetes-pf-flowschema-uid",
              "value": "3e535193-da03-4be0-8af9-31b77c23de3e"
            },
            {
              "name": "x-kubernetes-pf-prioritylevel-uid",
              "value": "1a41d4ba-2c5d-4828-b199-fd4ff13a2398"
            },
            {
              "name": "date",
              "value": "Fri, 12 Jan 2024 17:13:39 GMT"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "transfer-encoding",
              "value": "chunked"
            }
          ],
          "headersSize": 343,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2024-01-12T17:13:39.595Z",
        "time": 22,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 22
        }
      },
      {
        "_id": "f966932dcc2f7fa1fa3030bbee5bdf0d",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "name": "accept",
              "value": "application/json"
            },
            {
              "name": "host",
              "value": "192.168.49.2:8443"
            }
          ],
          "headersSize": 158,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "limit",
              "value": "250"
            }
          ],
          "url": "https://192.168.49.2:8443/apis/networking.k8s.io/v1/namespaces/default/networkpolicies?limit=250"
        },
        "response": {
          "bodySize": 1198,
          "content": {
            "mimeType": "application/json",
            "size": 1198,
            "text": "{\"kind\":\"NetworkPolicyList\",\"apiVersion\":\"networking.k8s.io/v1\",\"metadata\":{\"resourceVersion\":\"51782\"},\"items\":[{\"metadata\":{\"name\":\"db-frontend-allow\",\"namespace\":\"default\",\"uid\":\"2fbcad2d-1572-4a03-ae00-4f2fbfd8a2e8\",\"resourceVersion\":\"27751\",\"generation\":1,\"creationTimestamp\":\"2024-01-11T21:55:36Z\",\"managedFields\":[{\"manager\":\"kubectl-create\",\"operation\":\"Update\",\"apiVersion\":\"networking.k8s.io/v1\",\"time\":\"2024-01-11T21:55:36Z\",\"fieldsType\":\"FieldsV1\",\"fieldsV1\":{\"f:spec\":{\"f:ingress\":{},\"f:podSelector\":{},\"f:policyTypes\":{}}}}]},\"spec\":{\"podSelector\":{\"matchLabels\":{\"app\":\"np-test-mariadb\"}},\"ingress\":[{\"ports\":[{\"protocol\":\"TCP\",\"port\":3306}],\"from\":[{\"podSelector\":{\"matchLabels\":{\"app\":\"np-test-wordpress\"}}}]}],\"policyTypes\":[\"Ingress\"]}},{\"metadata\":{\"name\":\"default-deny\",\"namespace\":\"default\",\"uid\":\"d122159c-13d4-46f4-898a-4b253dfeb297\",\"resourceVersion\":\"27717\",\"generation\":1,\"creationTimestamp\":\"2024-01-11T21:55:11Z\",\"managedFields\":[{\"manager\":\"kubectl-create\",\"operation\":\"Update\",\"apiVersion\":\"networking.k8s.io/v1\",\"time\":\"2024-01-11T21:55:11Z\",\"fieldsType\":\"FieldsV1\",\"fieldsV1\":{\"f:spec\":{\"f:policyTypes\":{}}}}]},\"spec\":{\"podSelector\":{},\"policyTypes\":[\"Ingress\"]}}]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "audit-id",
              "value": "70e469c1-6ddb-4655-a6f6-d7a31d76bbbe"
            },
            {
              "name": "cache-control",
              "value": "no-cache, private"
            },
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "x-kubernetes-pf-flowschema-uid",
              "value": "3e535193-da03-4be0-8af9-31b77c23de3e"
            },
            {
              "name": "x-kubernetes-pf-prioritylevel-uid",
              "value": "1a41d4ba-2c5d-4828-b199-fd4ff13a2398"
            },
            {
              "name": "date",
              "value": "Fri, 12 Jan 2024 17:13:39 GMT"
            },
            {
              "name": "content-length",
              "value": "1198"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 337,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2024-01-12T17:13:39.670Z",
        "time": 16,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 16
        }
      },
      {
        "_id": "386e311aa5778e493f3647082ba33d55",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "name": "accept",
              "value": "application/json"
            },
            {
              "name": "host",
              "value": "192.168.49.2:8443"
            }
          ],
          "headersSize": 166,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "limit",
              "value": "250"
            }
          ],
          "url": "https://192.168.49.2:8443/apis/networking.k8s.io/v1/namespaces/kube-node-lease/networkpolicies?limit=250"
        },
        "response": {
          "bodySize": 115,
          "content": {
            "mimeType": "application/json",
            "size": 115,
            "text": "{\"kind\":\"NetworkPolicyList\",\"apiVersion\":\"networking.k8s.io/v1\",\"metadata\":{\"resourceVersion\":\"51782\"},\"items\":[]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "audit-id",
              "value": "23cd3364-c8f4-4c92-ba56-8ae506835875"
            },
            {
              "name": "cache-control",
              "value": "no-cache, private"
            },
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "x-kubernetes-pf-flowschema-uid",
              "value": "3e535193-da03-4be0-8af9-31b77c23de3e"
            },
            {
              "name": "x-kubernetes-pf-prioritylevel-uid",
              "value": "1a41d4ba-2c5d-4828-b199-fd4ff13a2398"
            },
            {
              "name": "date",
              "value": "Fri, 12 Jan 2024 17:13:39 GMT"
            },
            {
              "name": "content-length",
              "value": "115"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 336,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2024-01-12T17:13:39.696Z",
        "time": 16,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 16
        }
      },
      {
        "_id": "e6fbbd1e40374081e9b4bcaa02b4e8dd",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "name": "accept",
              "value": "application/json"
            },
            {
              "name": "host",
              "value": "192.168.49.2:8443"
            }
          ],
          "headersSize": 162,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "limit",
              "value": "250"
            }
          ],
          "url": "https://192.168.49.2:8443/apis/networking.k8s.io/v1/namespaces/kube-public/networkpolicies?limit=250"
        },
        "response": {
          "bodySize": 115,
          "content": {
            "mimeType": "application/json",
            "size": 115,
            "text": "{\"kind\":\"NetworkPolicyList\",\"apiVersion\":\"networking.k8s.io/v1\",\"metadata\":{\"resourceVersion\":\"51782\"},\"items\":[]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "audit-id",
              "value": "a8b6a783-3faf-4d11-aef1-a4b330c6aa94"
            },
            {
              "name": "cache-control",
              "value": "no-cache, private"
            },
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "x-kubernetes-pf-flowschema-uid",
              "value": "3e535193-da03-4be0-8af9-31b77c23de3e"
            },
            {
              "name": "x-kubernetes-pf-prioritylevel-uid",
              "value": "1a41d4ba-2c5d-4828-b199-fd4ff13a2398"
            },
            {
              "name": "date",
              "value": "Fri, 12 Jan 2024 17:13:39 GMT"
            },
            {
              "name": "content-length",
              "value": "115"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 336,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2024-01-12T17:13:39.717Z",
        "time": 17,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 17
        }
      },
      {
        "_id": "f4d19e8f5d15c7c0decb22ffe99ea768",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "name": "accept",
              "value": "application/json"
            },
            {
              "name": "host",
              "value": "192.168.49.2:8443"
            }
          ],
          "headersSize": 162,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "limit",
              "value": "250"
            }
          ],
          "url": "https://192.168.49.2:8443/apis/networking.k8s.io/v1/namespaces/kube-system/networkpolicies?limit=250"
        },
        "response": {
          "bodySize": 115,
          "content": {
            "mimeType": "application/json",
            "size": 115,
            "text": "{\"kind\":\"NetworkPolicyList\",\"apiVersion\":\"networking.k8s.io/v1\",\"metadata\":{\"resourceVersion\":\"51782\"},\"items\":[]}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "audit-id",
              "value": "dee49b53-a255-439b-96e3-626cb3a3ac37"
            },
            {
              "name": "cache-control",
              "value": "no-cache, private"
            },
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "x-kubernetes-pf-flowschema-uid",
              "value": "3e535193-da03-4be0-8af9-31b77c23de3e"
            },
            {
              "name": "x-kubernetes-pf-prioritylevel-uid",
              "value": "1a41d4ba-2c5d-4828-b199-fd4ff13a2398"
            },
            {
              "name": "date",
              "value": "Fri, 12 Jan 2024 17:13:39 GMT"
            },
            {
              "name": "content-length",
              "value": "115"
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 336,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2024-01-12T17:13:39.745Z",
        "time": 15,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 15
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
