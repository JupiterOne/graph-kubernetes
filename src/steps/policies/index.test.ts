import { fetchNetworkPolicies } from '.';
import { createDataCollectionTest } from '../../../test/recording';
import { integrationConfig } from '../../../test/config';
import { Entities, Relationships } from '../constants';
import { RelationshipClass } from '@jupiterone/data-model';
import { fetchNamespaces } from '../namespaces';

describe('#fetchNetworkPolicies', () => {
  test('should collect data', async () => {
    await createDataCollectionTest({
      recordingName: 'fetchNetworkPolicies',
      recordingDirectory: __dirname,
      integrationConfig,
      stepFunctions: [fetchNamespaces, fetchNetworkPolicies],
      entitySchemaMatchers: [
        {
          _type: Entities.NETWORK_POLICY._type,
          matcher: {
            _class: ['Configuration'],
            schema: {
              additionalProperties: false,
              properties: {
                _type: { const: 'kube_network_policy' },
                _rawData: {
                  type: 'array',
                  items: { type: 'object' },
                },
                name: { type: 'string' },
                displayName: { type: 'string' },
                generation: { type: 'number' },
                deletionGracePeriodSeconds: { type: 'string' },
                resourceVersion: { type: 'string' },
                createdOn: { type: 'number' },
                policyTypes: {
                  type: 'array',
                  items: { type: 'string' },
                },
              },
            },
          },
        },
      ],
      relationshipSchemaMatchers: [
        {
          _type: Relationships.NAMESPACE_CONTAINS_NETWORK_POLICY._type,
          matcher: {
            schema: {
              properties: {
                _class: { const: RelationshipClass.CONTAINS },
                _type: { const: 'kube_namespace_contains_network_policy' },
              },
            },
          },
        },
      ],
    });
  });
});
