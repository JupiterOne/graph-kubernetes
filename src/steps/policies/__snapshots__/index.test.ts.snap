// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#fetchNetworkPolicies should collect data: jobState 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "Group",
      ],
      "_key": "868d4703-b338-4051-98e4-b65b4f0a0006",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": V1Namespace {
            "apiVersion": undefined,
            "kind": undefined,
            "metadata": V1ObjectMeta {
              "annotations": undefined,
              "creationTimestamp": 2024-01-11T15:50:15.000Z,
              "deletionGracePeriodSeconds": undefined,
              "deletionTimestamp": undefined,
              "finalizers": undefined,
              "generateName": undefined,
              "generation": undefined,
              "labels": Object {
                "kubernetes.io/metadata.name": "default",
              },
              "managedFields": Array [
                V1ManagedFieldsEntry {
                  "apiVersion": "v1",
                  "fieldsType": "FieldsV1",
                  "fieldsV1": Object {
                    "f:metadata": Object {
                      "f:labels": Object {
                        ".": Object {},
                        "f:kubernetes.io/metadata.name": Object {},
                      },
                    },
                  },
                  "manager": "kube-apiserver",
                  "operation": "Update",
                  "subresource": undefined,
                  "time": 2024-01-11T15:50:15.000Z,
                },
              ],
              "name": "default",
              "namespace": undefined,
              "ownerReferences": undefined,
              "resourceVersion": "39",
              "selfLink": undefined,
              "uid": "868d4703-b338-4051-98e4-b65b4f0a0006",
            },
            "spec": V1NamespaceSpec {
              "finalizers": Array [
                "kubernetes",
              ],
            },
            "status": V1NamespaceStatus {
              "conditions": undefined,
              "phase": "Active",
            },
          },
        },
      ],
      "_type": "kube_namespace",
      "active": undefined,
      "createdOn": 1704988215000,
      "deletionGracePeriodSeconds": undefined,
      "displayName": "default",
      "finalizers": Array [
        "kubernetes",
      ],
      "generation": undefined,
      "name": "default",
      "resourceVersion": "39",
      "status.phase": "Active",
    },
    Object {
      "_class": Array [
        "Group",
      ],
      "_key": "52615a92-75c3-4c4b-8b72-51be82a4bc21",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": V1Namespace {
            "apiVersion": undefined,
            "kind": undefined,
            "metadata": V1ObjectMeta {
              "annotations": undefined,
              "creationTimestamp": 2024-01-11T15:50:15.000Z,
              "deletionGracePeriodSeconds": undefined,
              "deletionTimestamp": undefined,
              "finalizers": undefined,
              "generateName": undefined,
              "generation": undefined,
              "labels": Object {
                "kubernetes.io/metadata.name": "kube-node-lease",
              },
              "managedFields": Array [
                V1ManagedFieldsEntry {
                  "apiVersion": "v1",
                  "fieldsType": "FieldsV1",
                  "fieldsV1": Object {
                    "f:metadata": Object {
                      "f:labels": Object {
                        ".": Object {},
                        "f:kubernetes.io/metadata.name": Object {},
                      },
                    },
                  },
                  "manager": "kube-apiserver",
                  "operation": "Update",
                  "subresource": undefined,
                  "time": 2024-01-11T15:50:15.000Z,
                },
              ],
              "name": "kube-node-lease",
              "namespace": undefined,
              "ownerReferences": undefined,
              "resourceVersion": "35",
              "selfLink": undefined,
              "uid": "52615a92-75c3-4c4b-8b72-51be82a4bc21",
            },
            "spec": V1NamespaceSpec {
              "finalizers": Array [
                "kubernetes",
              ],
            },
            "status": V1NamespaceStatus {
              "conditions": undefined,
              "phase": "Active",
            },
          },
        },
      ],
      "_type": "kube_namespace",
      "active": undefined,
      "createdOn": 1704988215000,
      "deletionGracePeriodSeconds": undefined,
      "displayName": "kube-node-lease",
      "finalizers": Array [
        "kubernetes",
      ],
      "generation": undefined,
      "name": "kube-node-lease",
      "resourceVersion": "35",
      "status.phase": "Active",
    },
    Object {
      "_class": Array [
        "Group",
      ],
      "_key": "7a2ee4f7-12fd-494b-93f0-c639b1e63a72",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": V1Namespace {
            "apiVersion": undefined,
            "kind": undefined,
            "metadata": V1ObjectMeta {
              "annotations": undefined,
              "creationTimestamp": 2024-01-11T15:50:15.000Z,
              "deletionGracePeriodSeconds": undefined,
              "deletionTimestamp": undefined,
              "finalizers": undefined,
              "generateName": undefined,
              "generation": undefined,
              "labels": Object {
                "kubernetes.io/metadata.name": "kube-public",
              },
              "managedFields": Array [
                V1ManagedFieldsEntry {
                  "apiVersion": "v1",
                  "fieldsType": "FieldsV1",
                  "fieldsV1": Object {
                    "f:metadata": Object {
                      "f:labels": Object {
                        ".": Object {},
                        "f:kubernetes.io/metadata.name": Object {},
                      },
                    },
                  },
                  "manager": "kube-apiserver",
                  "operation": "Update",
                  "subresource": undefined,
                  "time": 2024-01-11T15:50:15.000Z,
                },
              ],
              "name": "kube-public",
              "namespace": undefined,
              "ownerReferences": undefined,
              "resourceVersion": "20",
              "selfLink": undefined,
              "uid": "7a2ee4f7-12fd-494b-93f0-c639b1e63a72",
            },
            "spec": V1NamespaceSpec {
              "finalizers": Array [
                "kubernetes",
              ],
            },
            "status": V1NamespaceStatus {
              "conditions": undefined,
              "phase": "Active",
            },
          },
        },
      ],
      "_type": "kube_namespace",
      "active": undefined,
      "createdOn": 1704988215000,
      "deletionGracePeriodSeconds": undefined,
      "displayName": "kube-public",
      "finalizers": Array [
        "kubernetes",
      ],
      "generation": undefined,
      "name": "kube-public",
      "resourceVersion": "20",
      "status.phase": "Active",
    },
    Object {
      "_class": Array [
        "Group",
      ],
      "_key": "ce86116b-863b-4e74-979e-3c031a9aeee1",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": V1Namespace {
            "apiVersion": undefined,
            "kind": undefined,
            "metadata": V1ObjectMeta {
              "annotations": undefined,
              "creationTimestamp": 2024-01-11T15:50:15.000Z,
              "deletionGracePeriodSeconds": undefined,
              "deletionTimestamp": undefined,
              "finalizers": undefined,
              "generateName": undefined,
              "generation": undefined,
              "labels": Object {
                "kubernetes.io/metadata.name": "kube-system",
              },
              "managedFields": Array [
                V1ManagedFieldsEntry {
                  "apiVersion": "v1",
                  "fieldsType": "FieldsV1",
                  "fieldsV1": Object {
                    "f:metadata": Object {
                      "f:labels": Object {
                        ".": Object {},
                        "f:kubernetes.io/metadata.name": Object {},
                      },
                    },
                  },
                  "manager": "kube-apiserver",
                  "operation": "Update",
                  "subresource": undefined,
                  "time": 2024-01-11T15:50:15.000Z,
                },
              ],
              "name": "kube-system",
              "namespace": undefined,
              "ownerReferences": undefined,
              "resourceVersion": "12",
              "selfLink": undefined,
              "uid": "ce86116b-863b-4e74-979e-3c031a9aeee1",
            },
            "spec": V1NamespaceSpec {
              "finalizers": Array [
                "kubernetes",
              ],
            },
            "status": V1NamespaceStatus {
              "conditions": undefined,
              "phase": "Active",
            },
          },
        },
      ],
      "_type": "kube_namespace",
      "active": undefined,
      "createdOn": 1704988215000,
      "deletionGracePeriodSeconds": undefined,
      "displayName": "kube-system",
      "finalizers": Array [
        "kubernetes",
      ],
      "generation": undefined,
      "name": "kube-system",
      "resourceVersion": "12",
      "status.phase": "Active",
    },
    Object {
      "_class": Array [
        "Configuration",
      ],
      "_key": "2fbcad2d-1572-4a03-ae00-4f2fbfd8a2e8",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": V1NetworkPolicy {
            "apiVersion": undefined,
            "kind": undefined,
            "metadata": V1ObjectMeta {
              "annotations": undefined,
              "creationTimestamp": 2024-01-11T21:55:36.000Z,
              "deletionGracePeriodSeconds": undefined,
              "deletionTimestamp": undefined,
              "finalizers": undefined,
              "generateName": undefined,
              "generation": 1,
              "labels": undefined,
              "managedFields": Array [
                V1ManagedFieldsEntry {
                  "apiVersion": "networking.k8s.io/v1",
                  "fieldsType": "FieldsV1",
                  "fieldsV1": Object {
                    "f:spec": Object {
                      "f:ingress": Object {},
                      "f:podSelector": Object {},
                      "f:policyTypes": Object {},
                    },
                  },
                  "manager": "kubectl-create",
                  "operation": "Update",
                  "subresource": undefined,
                  "time": 2024-01-11T21:55:36.000Z,
                },
              ],
              "name": "db-frontend-allow",
              "namespace": "default",
              "ownerReferences": undefined,
              "resourceVersion": "27751",
              "selfLink": undefined,
              "uid": "2fbcad2d-1572-4a03-ae00-4f2fbfd8a2e8",
            },
            "spec": V1NetworkPolicySpec {
              "egress": undefined,
              "ingress": Array [
                V1NetworkPolicyIngressRule {
                  "from": Array [
                    V1NetworkPolicyPeer {
                      "ipBlock": undefined,
                      "namespaceSelector": undefined,
                      "podSelector": V1LabelSelector {
                        "matchExpressions": undefined,
                        "matchLabels": Object {
                          "app": "np-test-wordpress",
                        },
                      },
                    },
                  ],
                  "ports": Array [
                    V1NetworkPolicyPort {
                      "endPort": undefined,
                      "port": 3306,
                      "protocol": "TCP",
                    },
                  ],
                },
              ],
              "podSelector": V1LabelSelector {
                "matchExpressions": undefined,
                "matchLabels": Object {
                  "app": "np-test-mariadb",
                },
              },
              "policyTypes": Array [
                "Ingress",
              ],
            },
          },
        },
      ],
      "_type": "kube_network_policy",
      "createdOn": 1705010136000,
      "deletionGracePeriodSeconds": undefined,
      "displayName": "db-frontend-allow",
      "generation": 1,
      "name": "db-frontend-allow",
      "policyTypes": Array [
        "Ingress",
      ],
      "resourceVersion": "27751",
    },
    Object {
      "_class": Array [
        "Configuration",
      ],
      "_key": "d122159c-13d4-46f4-898a-4b253dfeb297",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": V1NetworkPolicy {
            "apiVersion": undefined,
            "kind": undefined,
            "metadata": V1ObjectMeta {
              "annotations": undefined,
              "creationTimestamp": 2024-01-11T21:55:11.000Z,
              "deletionGracePeriodSeconds": undefined,
              "deletionTimestamp": undefined,
              "finalizers": undefined,
              "generateName": undefined,
              "generation": 1,
              "labels": undefined,
              "managedFields": Array [
                V1ManagedFieldsEntry {
                  "apiVersion": "networking.k8s.io/v1",
                  "fieldsType": "FieldsV1",
                  "fieldsV1": Object {
                    "f:spec": Object {
                      "f:policyTypes": Object {},
                    },
                  },
                  "manager": "kubectl-create",
                  "operation": "Update",
                  "subresource": undefined,
                  "time": 2024-01-11T21:55:11.000Z,
                },
              ],
              "name": "default-deny",
              "namespace": "default",
              "ownerReferences": undefined,
              "resourceVersion": "27717",
              "selfLink": undefined,
              "uid": "d122159c-13d4-46f4-898a-4b253dfeb297",
            },
            "spec": V1NetworkPolicySpec {
              "egress": undefined,
              "ingress": undefined,
              "podSelector": V1LabelSelector {
                "matchExpressions": undefined,
                "matchLabels": undefined,
              },
              "policyTypes": Array [
                "Ingress",
              ],
            },
          },
        },
      ],
      "_type": "kube_network_policy",
      "createdOn": 1705010111000,
      "deletionGracePeriodSeconds": undefined,
      "displayName": "default-deny",
      "generation": 1,
      "name": "default-deny",
      "policyTypes": Array [
        "Ingress",
      ],
      "resourceVersion": "27717",
    },
  ],
  "collectedRelationships": Array [
    Object {
      "_class": "CONTAINS",
      "_fromEntityKey": "868d4703-b338-4051-98e4-b65b4f0a0006",
      "_key": "868d4703-b338-4051-98e4-b65b4f0a0006|contains|2fbcad2d-1572-4a03-ae00-4f2fbfd8a2e8",
      "_toEntityKey": "2fbcad2d-1572-4a03-ae00-4f2fbfd8a2e8",
      "_type": "kube_namespace_contains_network_policy",
      "displayName": "CONTAINS",
    },
    Object {
      "_class": "CONTAINS",
      "_fromEntityKey": "868d4703-b338-4051-98e4-b65b4f0a0006",
      "_key": "868d4703-b338-4051-98e4-b65b4f0a0006|contains|d122159c-13d4-46f4-898a-4b253dfeb297",
      "_toEntityKey": "d122159c-13d4-46f4-898a-4b253dfeb297",
      "_type": "kube_namespace_contains_network_policy",
      "displayName": "CONTAINS",
    },
  ],
  "encounteredTypes": Array [
    "kube_namespace",
    "kube_network_policy",
    "kube_namespace_contains_network_policy",
  ],
  "numCollectedEntities": 6,
  "numCollectedRelationships": 2,
}
`;
